lms/djangoapps/courseware/module_render.py

```diff
@@ -165,6 +171,10 @@ def toc_for_course(request, course, active_chapter, active_section, field_data_c
                 if unicode(chapter.location) not in required_content:
                     local_hide_from_toc = True
 
+            # check if user is allowed to skip entrance exam then don't hide content
```

Comment:
this check should not be needed after moving skip entrance exam login into `_get_required_content()`